### PR TITLE
Scala eventsourced counter example

### PR DESCRIPTION
**Pull request by RayRoestenburg**
Includes some fixes for scala event sourced entity (including https://github.com/lightbend/akkaserverless-java-sdk/pull/563 which you likely already reviewed).

### CODE REVIEW

File: codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/EventSourcedEntitySourceGeneratorSuite.scala [hunk 1]
```diff
@@ -111,7 +111,7 @@ class EventSourcedEntitySourceGeneratorSuite extends munit.FunSuite {
           |class MyEntityHandler(entity: MyEntity) extends EventSourcedEntityHandler[MyState, MyEntity](entity) {
           |  def handleEvent(state: MyState, event: Any): MyState = {
           |    event match {
-          |      case SetEvent =>
+          |      case _: SetEvent =>
```
Comment (octonato) [inline codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/EventSourcedEntitySourceGeneratorSuite.scala:114]: oh, here we can do

```scala
case evt: SetEvent => entity.setEvent(state, evt)
```

and remove the instance of :-)

Comment (octonato) [reply codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/EventSourcedEntitySourceGeneratorSuite.scala:114]: We have probably coded like that for the other components as well and we can clean up. Maybe later. Don't need to be now.

File: samples/scala-valueentity-counter/.scalafmt.conf [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  only
+  noneOf
+  oneElementOf
+  noElementsOf
+  atLeastOneElementOf
+  atMostOneElementOf
+  allElementsOf
+  inOrderElementsOf
+  theSameElementsAs
+]
```
Comment (octonato) [inline samples/scala-valueentity-counter/.scalafmt.conf:47]: Maybe better to add this on another PR and then also format the code in the valueentity-counter sample.

**Review by octonato — APPROVED** [review]
LGTM

File: codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/EventSourcedEntitySourceGeneratorSuite.scala (hunk 1 above)
Comment (RayRoestenburg) [reply codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/EventSourcedEntitySourceGeneratorSuite.scala:114]: argh I've become too mechanical LOL

**Pull request merged**